lib/evmone/instructions_calls.cpp

```diff
@@ -300,19 +300,17 @@ Result create_impl(StackTop stack, int64_t gas_left, ExecutionState& state) noex
     return {EVMC_SUCCESS, gas_left};
 }
 
-Result eofcreate(
+template <Opcode Op>
+Result create_eof_impl(
     StackTop stack, int64_t gas_left, ExecutionState& state, code_iterator& pos) noexcept
 {
+    static_assert(Op == OP_EOFCREATE || Op == OP_TXCREATE);
+
     if (state.in_static_mode())
         return {EVMC_STATIC_MODE_VIOLATION, gas_left};
 
-    const auto initcontainer_index = uint8_t{pos[1]};
-    pos += 2;
-
-    const auto& container = state.original_code;
-    const auto eof_header = read_valid_eof1_header(state.original_code);
-    const auto initcontainer = eof_header.get_container(container, initcontainer_index);
-
+    const auto initcode_hash =
```

Comment:
It's only used for `OP_TXCREATE`. Can we move it to `else` below?

Reply:
We need to pop the stack in the right order...